kernel/kevinfs.c

```diff
@@ -136,12 +137,22 @@ static int kevinfs_ata_write_superblock(struct device *device)
 	super.num_free_blocks = free_blocks;
 
 	memcpy(wbuffer, &super, sizeof(super));
-
-	uint32_t i;
-	for (i = super.inode_bitmap_start; i < super.free_block_start; i++) {
+	uint32_t counter = 0;
+	printf("Writing inode bitmap...\n");
+	for (uint32_t i = super.inode_bitmap_start; i < super.inode_start; i++) {
+		if(!device_write(device, zeros, 1, i))
+			return -1;
```

Comment:
Handling errors deep in the code is a perennial challenge.  Double check: Does the layer above this detect a failure to write and pass it up?

Reply:
That error code gets propagated up the stack to fs_mkfs, which gets called by kshell when formatting a disk (kshell.c line 201). `fs_mkfs` returns `KERROR_NOT_SUPPORTED`. However, the shell does not check the return value of this function. I'm guessing that fixing the shell should be a separate issue/PR though